utils/build-script-impl

```diff
@@ -1226,7 +1232,16 @@ if [ -e "${WORKSPACE}/compiler-rt" ] ; then
     fi
 fi
 
+# Build libcxx, unless it doesn't exist.
+LIBCXX_SOURCE_DIR="${WORKSPACE}/libcxx"
+if [[ ! -e "${LIBCXX_SOURCE_DIR}" ]] ; then
+  SKIP_BUILD_LIBCXX=1
+fi
+
 PRODUCTS=(cmark llvm)
+if [[ ! "${SKIP_BUILD_LIBCXX}" ]] ; then
+     PRODUCTS=("${PRODUCTS[@]}" libcxx)
+fi
```

Comment:
Because I turned on --no-legacy-impl by default, this won't be enough.  You need to add a `Product` for libcxx in `build-script` (the python code) so that this will be triggered.  Right now it will be skipped by the check for `$(should_execute_host_actions_for_phase ${host} build)` below.